src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -966,6 +966,7 @@ private boolean doPrepExec(TDSWriter tdsWriter,
             if (needsPrepare 
                 && !connection.getEnablePrepareOnFirstPreparedStatementCall() 
                 && !isExecutedAtLeastOnce
+                && preparedTypeDefinitions.length() > 0
```

Comment:
We should not make this check here. sp_executesql works fine without arguments. The problem is in the buildExecSQLParams method. Basically this  works fine sp_prepexec/buildPrepExecParams but not for sp_executesql:
`tdsWriter.writeRPCStringUnicode((preparedTypeDefinitions.length() > 0) ? preparedTypeDefinitions : null);`

So we should just change that to:
`if (preparedTypeDefinitions.length() > 0)
tdsWriter.writeRPCStringUnicode(preparedTypeDefinitions);`